### ISSUE TITLE
possible bug in Frontend Code

### ISSUE THREAD

**Issue by mlodic** [GreedyBear-Project/GreedyBear#71]
In the most recent version in `develop`, with the new GUI, there is a possible hidden bug in the rendering of the frontend code.

AppHeaders is rendered multiple times, maybe due to some problem in handling the status in React. Same things happen for other parts of the code in other sections. To be investigated (first step, check output in the console log)

**Comment by TacitTactics** [comment]
I get a blank page when I attempt to load greedybear

**Comment by mlodic** [comment]
please add more information, screenshots, logs,etc otherwise we can't help

**Comment by dodaa08** [comment]
Hi, I see the issue with multiple rendering, but for reaching there I need to fix some warnings shown in the console, even if i try to update the type of spinner to border or grow,  the warnings will have no effect, could you help approach this issue ?

![image](https://github.com/user-attachments/assets/4b0d07cb-2c00-436c-a580-4d5b0193fcec)

**Comment by drosetti** [comment]
Hi dodaa, are you sure that is the correct spinner ? One I found a similar warning related to a component imported by the certego-ui dependency. I suggest to check if it's the spinner in the code or in the dependency

**Comment by dodaa08** [comment]
Okay, I'll recheck and get back to you.

**Comment by dodaa08** [comment]
import { Spinner } from "reactstrap";  I think  it's imported from reactstrap throughout the project, So, is it because of the nested usage of both the libraries ?

> Hi dodaa, are you sure that is the correct spinner ? One I found a similar warning related to a component imported by the certego-ui dependency. I suggest to check if it's the spinner in the code or in the dependency

**Comment by drosetti** [comment]
ok, in this case it's imported from reactastrap. Yes, reactstrap and the spinner is used directly from intelOwl, but also in the certego-ui and it's not up-to-date and it could be the reason for this error. Even if in the jsx file you are viewing there is not the certego-ui import, it could be in a component that use the component you are inspecting.

**Comment by aditisingh70330** [comment]
@regulartim can you assign me this issue?

**Comment by regulartim** [comment]
Hey @aditisingh70330 ! What is the solution that you are proposing? Are you able to reproduce the issue?

**Comment by aditisingh70330** [comment]
@regulartim Hi! Yes, I was able to reproduce it on the latest 'develop'  branch.
When I load the UI, I sometimes get a blank page. While debugging, I added some console logs and noticed that 'AppHeaders'(and a few other components are rendering multiple times.From what I’ve checked so far, part of it is React StrictMode, but there are extra re-renders beyond that. It looks like something in state updates or 'useEffect' might be triggering them, and I also want to double-check that 'AppHeaders' isn’t being mounted more than once in the layout.Next, I’m going to trace where the extra renders are coming from and isolate the exact cause of the blank page (probably a runtime error or rendering issue). I should be able to narrow it down and push a fix soon.

**Comment by aditisingh70330** [comment]
@regulartim should i attempt to resolve this?

**Comment by regulartim** [comment]
Yes! Sorry, forgot to assign you.

**Comment by aditisingh70330** [comment]
The frontend is not showing any AppHeader rendering issue on the React side.
The multiple logs I observed are consistent with normal React dev behavior (including StrictMode / state initialization).
However, I am currently seeing repeated 502 Bad Gateway errors for /api/auth/authentication and /api/news, which suggests the backend service or proxy layer is not responding correctly.
I’ll verify whether the backend service is running and check the API gateway configuration, since this is likely causing the downstream UI issues.

**Comment by regulartim** [comment]
> The frontend is not showing any AppHeader rendering issue on the React side.
The multiple logs I observed are consistent with normal React dev behavior (including StrictMode / state initialization

So you can not reproduce the issue?

**Comment by aditisingh70330** [comment]
At the moment, I am consistently observing 502 Bad Gateway responses for /api/auth/authentication and /api/news, which suggests that the backend/API service is not reachable in the current setup. The frontend is making requests to /api/news and /api/auth, but these endpoints are returning 502 Bad Gateway, indicating that the backend service is currently not reachable or not running. Overall, the frontend rendering logic looks stable from what I can observe.

**Comment by aditisingh70330** [comment]
I’m trying to locate the backend part of the project—could you guide me?

**Comment by regulartim** [comment]
> I’m trying to locate the backend part of the project—could you guide me?

I am not sure what you mean with "locate". You have to start the application via `docker compose` to get the backend up. Please take a look at the [wiki](https://github.com/GreedyBear-Project/GreedyBear/wiki) for further information.

**Comment by aditisingh70330** [comment]
Hi! I’ve worked on the issue you assigned (#71) and implemented a fix for the frontend rendering problem.

* Removed unnecessary FallBackLoading usage
* Fixed multiple rendering/render duplication issue
* Verified application runs correctly without errors

Tested locally and confirmed fix works as expected.

**Comment by regulartim** [comment]
@aditisingh70330 , how would you suggest to continue here after merging #1302 ?

**Comment by aditisingh70330** [comment]
@regulartim Good question  
I’d keep the other errors separate from #71 unless we find a direct link.

From what I’ve seen so far:

- The 502 errors is a backend / docker setup issue rather than something in the frontend rendering.
- The spinner warning (type="ripple") seems to come from the shared UI dependency, not code in this repo.
- The other warnings (Fade/refs, router, zustand) look more like general UI/dependency issues.

So for now I’d focus #71 just on confirmed frontend rendering/remounting behavior.

After that, I’m happy to:
- help debug the 502 issue if needed, or
- open small separate PRs for any frontend warnings that actually come from this repo.

Let me know if you’d prefer grouping any of those into this issue instead.

**Comment by aditisingh70330** [comment]
I re-tested the latest develop after #1302 across Home, Dashboard, and Feeds.

From the logs, AppHeader is no longer continuously remounting. I only see normal route-related renders, not a repeated render loop.

The remaining console issues seem separate:
- 502 responses from API endpoints, likely backend/docker related
- Spinner/FallBackLoading warning from the shared UI dependency
- Fade/ref warnings in dashboard UI components
- one unmounted state update warning around AttackOriginMap, which might be worth investigating separately

So I don’t see another confirmed AppHeader remounting cause right now. I can continue by checking the AttackOriginMap unmounted state update warning as a separate focused follow-up if that sounds useful.

**Comment by regulartim** [comment]
Thanks for the investigation @aditisingh70330 ! You probably see the 502s because you don't have a backend running. For the other problems you found, please open (a) separate issue(s) if they are reproducible.
